### ISSUE TITLE
AAD is not configured message shown on the SignUp page when AAD is configured

### ISSUE THREAD

**Issue by jonasanker** [Azure/api-management-developer-portal#1686]
**Every bug report should have precise description and reproduction steps; console traces or source code references are appreciated.**

**For assistance requests, contact Azure support or submit a post [on Stack Overflow](http://aka.ms/apimso). We don't provide support through GitHub Issues. Feature requests can be raised [on the Azure Feedback Forum](https://aka.ms/apimwish).**

## Bug description

An Azure Active Directory is not configured message is shown on the Signup page when Azure Active Directory is enabled and working. 

Message
```
This widget will display a sign-up form when you configure Azure Active Directory or Azure Active Directory B2C integration in your API Management service. This message appears only in the portal's administrative mode and the widget will be rendered as an empty space in the published portal, so you don't need to remove it.
```

## Reproduction steps

1. Setup Azure Active Directory in APIM
2. Signup with a user who is part of the AD but have not previously signed up
3. See error at the bottom of the Sign Up page

## Expected behavior

The message is only shown when in administrative mode or when AAD/B2C is not configured

## Is your portal managed or self-hosted?

Managed

## Release tag or commit SHA (if using self-hosted version)

N/A

## API Management service name

Bug is present on several service instances, not isolated to one.

## Environment

 - Operating system: Windows 10
 - Browser: Google Chrome v99.0.4844.51, Edge 99.0.1150.36
 - Version: See above

## Additional context

Screenshot
![screenshot](https://user-images.githubusercontent.com/25930103/157404452-e20373ee-f6ab-4e8f-af2b-c3f51414f2ee.png)

**Comment by jonasanker** [comment]
It seems a similar fix for the Sign In page was made in #1621 but was not applied on the SignUp page. 

My typescript/knockout-js is sparse but I suspect this might relate to the type comparison error in [signupSocial.html](https://github.com/Azure/api-management-developer-portal/blob/master/src/components/users/signup-social/ko/signupSocial.html): 
```html
<!-- ko ifnot: identityProvider && (mode === "publishing") -->
```
where `mode` is an Observable and not a string.

**Comment by jonasanker** [comment]
@azaslonov I have added a suggestion for fixing the bug in #1687 which you can take in as you desire 🙂
